### DISCUSSION TITLE
Networked animation state?

### DISCUSSION THREAD

**Discussion by jameskane05** [mozilla/hubs#2956] [General Questions & Discussion] [answered]
What would it take to network/sync animation state in the same way that video player state is synced - so that all connected clients are seeing more or less the same keyframes at the same time? As I understand it, video players are synced across clients by assigning a user as "timekeeper" - can this be done for looping, auto-playing animations, as well?

**Comment by rawnsley** [comment]
Related https://github.com/mozilla/hubs/issues/2529

**Comment by netpro2k** [comment, marked as answer]
Yep, you are correct in how video players are synced. One client is selected as the "owner" of that video, and they periodically sync time out to other clients. If one of the non-owner clients detects it is sufficiently deviated (we allow up to a few seconds) from the timekeeper they will seek their video to the correct time.

Internally all the 3d model animations are played back using an `AnimationMixer` in ThreeJS https://threejs.org/docs/#api/en/animation/AnimationMixer, it has `time` property you can query and a `setTime` method you can use to jump to a specific time. This is basically the only interface you need to do something similar, though it would definitely be some work. You would probably also need to sync over which animation clips were currently playing (we do this part for hand animations, but don't bother trying to sync the time).
